bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,542 @@
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+
+import com.daveoxley.cbus.CGateConnectException;
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.CGateInterface;
+import com.daveoxley.cbus.CGateSession;
+import com.daveoxley.cbus.CGateThreadPool;
+import com.daveoxley.cbus.CGateThreadPoolExecutor;
+import com.daveoxley.cbus.events.EventCallback;
+import com.daveoxley.cbus.status.StatusChangeCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+
+/**
+ * The {@link CBusThreadPool} is responsible for executing jobs from a threadpool
+ *
+ * @author John Harvey - Initial contribution
+ */
+
+@NonNullByDefault
+final class CBusThreadPool extends CGateThreadPool
+{
+    @NonNullByDefault
+    public class CBusThreadPoolExecutor extends CGateThreadPoolExecutor
+    {
+        private final ThreadPoolExecutor threadPool;
+                
+        public CBusThreadPoolExecutor(@Nullable String poolName)
+        {
+            threadPool = (ThreadPoolExecutor) ThreadPoolManager.getPool(poolName);
+        }
+        
+        @Override
+        protected void execute(@Nullable Runnable runnable)
+        {
+            threadPool.execute(runnable);
+        }
+    }
+    public CBusThreadPool()
+    {}
+    private Map<@Nullable String, CGateThreadPoolExecutor> executorMap= new HashMap<@Nullable String, CGateThreadPoolExecutor>();
+
+    @Override
+    @SuppressWarnings({ "unused", "null" }) /* map.get() can return null */
+    protected  CGateThreadPoolExecutor CreateExecutor(@Nullable String name)
+    {
+        @Nullable CGateThreadPoolExecutor executor = executorMap.get(name);
+
+        if (executor != null)
+        {
+            return executor;
+        }
+        CGateThreadPoolExecutor newExecutor =  new CBusThreadPoolExecutor(name);
+        executorMap.put(name, newExecutor);
+        return newExecutor;
+    }
+}
+
+/**
+ * The {@link CBusCGateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+
+@NonNullByDefault
+public class CBusCGateHandler extends BaseBridgeHandler {
```

Comment:
please run the spotless formatter on this file

Reply:
Ran spotless formatter on all files